js/src/components/app-input-control/index.js

```diff
@@ -10,11 +10,26 @@ import { __experimentalInputControl as InputControl } from '@wordpress/component
  */
 import './index.scss';
 
+const baseClassName = 'app-input-control';
+
+/**
+ * Renders <InputControl> with a wrapper to be applicable extend additional class names.
+ * noPointerEvents
+ *
+ * @param {Object} props React props to be forwarded to {@link InputControl}.
+ * @param {string} [props.className] Additional CSS class name to be appended.
+ * @param {boolean} [props.noPointerEvents] Whether disabled all pointer events. It will attach `pointer-events: none;` onto wrapper if true.
+ */
 const AppInputControl = forwardRef( ( props, ref ) => {
-	const { className, ...rest } = props;
+	const { className, noPointerEvents, ...rest } = props;
+	const wrapperClassName = classnames(
+		baseClassName,
+		noPointerEvents && `${ baseClassName }--no-pointer-events`,
```

Comment:
❔ I think we don't need `noPointerEvents`, we can just rely on `readOnly` props? 🤔

Reply:
Not sure your meaning. Relying on `readOnly ` will still have the problem you reviewed in  https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r813123513.

Reply:
Sorry, I should have been clearer that this is related to https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r814105386. I hope the following illustrates better, we use the `readOnly` prop, add the "read-only" CSS class, and for the "read-only" class, we say no pointer events: 

```suggestion
		readOnly && `${ baseClassName }--read-only`,
```

And then in the CSS: 

```
	// For read-only elements, we don't want pointer events.
	// (And the read-only element is open to other CSS modifications like color, text size etc.
	&--read-only {
		pointer-events: none;
	}
```

Reply:
Thanks for the explanation!

The following reply also includes replies to [another comment](https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r815028101).

```js
readOnly && `${ baseClassName }--read-only`
```

I don't think it's a good idea to force binding the `readonly` attribute and `pointer-events: none` style inside an elementary component. This will decrease the flexibility of its use.

> For example, if we want a read-only element to have no pointer events, display text in yellow color and with red background, the class naming "read-only" would be better than "no-pointer-events".

In addition, the naming of `--read-only` also brings an ambiguous meaning because they are different.
- A **\<input\>** with `readonly` attribute still works with pointer events.
- A **\<input\>** with `pointer-events: none` style is still editable.

if there're other styles should be applied when setting `pointer-events: none` on elements, then I would still name it `&--no-pointer-events`. And class having other styles along with `readonly` attribute would be `&--read-only`, `&:read-only` or `&[readonly]` then.

> The reason is to make the CSS class describe what it is, instead of how it is supposed to look like.

`pointer-events: none` doesn't change the appearance of <InputSelect>, it changes the way it interacts, it's a style but acts more like an attribute. I don't understand why you think that the naming is closer to "how it is supposed to look like". If want a more precise description of what it is, `--no-pointer-events` should be a precise description already. Or what about `--ignore-pointer-events` or `--disable-pointer-events` instead? Or do you have an ideal naming without above the ambiguity?

Reply:
>> The reason is to make the CSS class describe what it is, instead of how it is supposed to look like.

> I don't understand why you think that the naming is closer to "how it is supposed to look like".

I'll slightly rephrase my original sentence to this: "The reason is to make the CSS class describe what it is semantically (that is generally understandable, by a developer or even a user), instead of how it is supposed to look like or act like in technical terms."

For example, a div containing error message would have class like ".error", warning message with ".warning", dangerous destructive button with ".danger". Those CSS class describe what they are semantically. Then we would have technical specific styling (text color, background color, border etc) for those CSS class in CSS files. 

> In addition, the naming of `--read-only` also brings an ambiguous meaning because they are different.
> 
> * A **<input>** with `readonly` attribute still works with pointer events.
> * A **<input>** with `pointer-events: none` style is still editable.

The way I see "read-only" here is not in the technical sense but in the semantical sense. When we say the term "read only" (like "error", "warning", "disabled" etc), likely everyone can understand the meaning of it. However, we can't really say the same with "no-pointer-events".

> if there're other styles should be applied when setting `pointer-events: none` on elements, then I would still name it `&--no-pointer-events`.

I take it that you mean it like this?

```css
.no-pointer-events { /* I simplify the class name here. */
	pointer-events: none;

	/* other styles... */
	color: red;
	background-color: pink;
	border: solid 1px red;
	/* ... and possibly many other styles. */
}
```

If that's the case, I find it difficult / weird to wrap my head around it, e.g. a `<div class="no-pointer-events">` with those UI styles applied to it kinda goes against my intuition. 😅 

> Or what about `--ignore-pointer-events` or `--disable-pointer-events` instead?

Those aren't very much different from `no-pointer-events`. I see "pointer events" as a technical implementation term and not semantic term. It is like `<div class="green-text black-background disable-pointer-events">`, with class names indicating implementation; I would prefer it like `<div class="matrixbox">` and have the technical implementation inside `.matrixbox { ... }` CSS.

---

I hope I get my points across. If you prefer your current implementation over my suggestions comments above, feel free to go ahead and merge, it is not a blocking issue and I don't want to hold up the project on this. 😄 🙏

Reply:
> [...], instead of how it is supposed to look like or act like in technical terms."

Sorry, I don't get why a code or a name in codes for readers, who are mostly devs, is not supposed to look like a technical term? I think it depends more on the abstraction level or readability (under a lesser ambiguity).

> For example, a div containing error message would have class like ".error", warning message with ".warning", dangerous destructive button with ".danger". Those CSS class describe what they are semantically. Then we would have technical specific styling (text color, background color, border etc) for those CSS class in CSS files.

I agree with this. And even the `&--no-pointer-events` is exactly fit with the semantical need because `pointer-events: none` only affect the behavior and doesn't change any appearance of \<InputSelect\> nor make something looks like read-only.

> If that's the case, I find it difficult / weird to wrap my head around it, e.g. a `<div class="no-pointer-events">` with those UI styles applied to it kinda goes against my intuition.

If that makes it difficult to understand, then I will use `&--no-pointer-events-with-what-it-is-semantically-style` such as `&--no-pointer-events-with-disabled-style`. Or separate it to `&--no-pointer-events` and `&--disabled-entire-selector`.

> I would prefer it like `<div class="matrixbox">` and have the technical implementation inside `.matrixbox { ... }` CSS.

😕  Sorry, I really don't get why is matrixbox. When seeing a `.matrixbox` CSS class, how could it be easily and accurately deduced the class will disable point events?

---

> I hope I get my points across. If you prefer your current implementation over my suggestions comments above, feel free to go ahead and merge, it is not a blocking issue and I don't want to hold up the project on this.

Thanks for the explanation!

But sorry I still feel confused. 🙏 
I will proceed with merging. If you think this implementation is not applicable from some perspectives, I'd be happy to discuss it separately later, or have you improve it in another PR. 🙇‍♂️